src/caselawclient/search_result.py

```diff
@@ -0,0 +1,264 @@
+    assigned_to: str = ""
+    editor_hold: str = "false"
+    editor_priority: str = ""
+    last_modified: str = ""
+
+    @property
+    def editor_status(
+        self,
+    ) -> EditorStatus:
+        """
+        Get the editor status based on the metadata.
+
+        Returns:
+            EditorStatus: The editor status.
+        """
+        if self.editor_hold == "true":
+            return EditorStatus.HOLD
+        if self.assigned_to:
+            return EditorStatus.IN_PROGRESS
+        return EditorStatus.NEW
+
+    @staticmethod
+    def create_from_uri(uri: str) -> "SearchResultMeta":
+        """
+        Create a SearchResultMeta instance from a URI.
+
+        Args:
+            uri (str): The URI of the search result.
+
+        Returns:
+            SearchResultMeta: The created SearchResultMeta instance.
+        """
+        response_text = api_client.get_properties_for_search_results([uri])
+        last_modified = api_client.get_last_modified(uri)
+        root = etree.fromstring(response_text)
+        return SearchResultMeta.create_from_node(root, last_modified)
+
+    @staticmethod
+    def create_from_node(
+        node: etree._Element, last_modified: str
+    ) -> "SearchResultMeta":
+        """
+        Create a SearchResultMeta instance from an XML node.
+
+        Args:
+            node (etree._Element): The XML node representing the search result.
+            last_modified (str): The last modified date.
+
+        Returns:
+            SearchResultMeta: The created SearchResultMeta instance.
+        """
+        extracted_submission_datetime = (
+            one(node.xpath("//transfer-received-at/text()")) or ""
+        )
+        submission_datetime = (
+            datetime.strptime(extracted_submission_datetime, "%Y-%m-%dT%H:%M:%SZ")
+            if extracted_submission_datetime
+            else datetime.min
+        )
+        # we assume that there is only one node in the search result XML.
+        return SearchResultMeta(
+            author=one(node.xpath("//source-name/text()")) or "",
+            author_email=one(node.xpath("//source-email/text()")) or "",
+            consignment_reference=one(
+                node.xpath("//transfer-consignment-reference/text()")
+            )
+            or "",
+            submission_datetime=submission_datetime,
+            assigned_to=one(node.xpath("//assigned-to/text()")) or "",
+            editor_hold=one(node.xpath("//editor-hold/text()")) or "",
+            editor_priority=one(node.xpath("//editor-priority/text()"))
+            or "20",  # medium priority
+            last_modified=last_modified,
+        )
+
+
+def one(x: Any) -> Optional[Any]:
+    """
+    Return the first item in the given iterable.
+
+    Args:
+        x (Any): The iterable.
+
+    Returns:
+        Optional[Any]: The first item in the iterable or None if the iterable is empty.
+    """
+    if len(x) > 1:
+        raise RuntimeError("More than one found.")
```

Comment:
this function was taken straight from PUI and EUI as well.
as was  the comment `# we assume that there is only one node in the search result XML.` on line  233

This runtimerror line is not currently covered by any tests... 

I will probably rename and rewite this function so that it is clearer that it is concerned with when an xpath query returns more than 1 result.

Then will write a test for it

Reply:
removed this function and refactored all the xpath code to simplify